src/WeightTrackerSecondary.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Objects;
+
+import components.map.Map;
+import components.sequence.Sequence;
+
+public abstract class WeightTrackerSecondary extends Object
```

Comment:
Object is implied and does not need to be explicitly listed. 
```suggestion
public abstract class WeightTrackerSecondary
```